cli/tools/upgrade.rs

```diff
@@ -387,55 +384,60 @@ fn check_exe(exe_path: &Path) -> Result<(), AnyError> {
   Ok(())
 }
 
-fn parse_upgrade_check_file(
-  content: String,
-) -> (Option<String>, Option<String>) {
-  let (mut last_checked, mut latest_version) = (None, None);
-  let split_content = content.split('!').collect::<Vec<_>>();
+struct CheckVersionFile {
+  pub last_checked: chrono::DateTime<chrono::Utc>,
+  pub latest_version: String,
+}
 
-  if split_content.len() == 2 {
-    last_checked = Some(split_content[0].to_owned());
+impl CheckVersionFile {
+  pub fn parse(content: String) -> Option<Self> {
+    let split_content = content.split('!').collect::<Vec<_>>();
 
-    if !split_content[1].is_empty() {
-      latest_version = Some(split_content[1].to_owned());
+    if split_content.len() != 2 {
+      return None;
     }
+    let last_checked = chrono::DateTime::parse_from_rfc3339(&split_content[0])
+      .map(|dt| dt.with_timezone(&chrono::Utc))
+      .ok()?;
+    let latest_version = split_content[1].to_owned();
+
+    Some(CheckVersionFile {
+      last_checked,
+      latest_version,
+    })
   }
 
-  (last_checked, latest_version)
+  pub fn serialize(&self) -> String {
+    format!("{}!{}", self.last_checked.to_rfc3339(), self.latest_version)
+  }
 }
 
-#[test]
-fn test_parse_upgrade_check_file() {
-  let (last_checked, latest_version) =
-    parse_upgrade_check_file("2020-01-01T00:00:00+00:00!1.2.3".to_string());
-  assert_eq!(last_checked, Some("2020-01-01T00:00:00+00:00".to_string()));
-  assert_eq!(latest_version, Some("1.2.3".to_string()));
-
-  let (last_checked, latest_version) =
-    parse_upgrade_check_file("2020-01-01T00:00:00+00:00!".to_string());
-  assert_eq!(last_checked, Some("2020-01-01T00:00:00+00:00".to_string()));
-  assert_eq!(latest_version, None);
```

Comment:
Not a realistic scenario, so I removed this test.

Reply:
Users can fiddle with the file 🤷

Reply:
Sure, I'll add it back, but it would return `None` for both now.